### PR TITLE
Change pytz to timezone.utc and ZoneInfo

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
This PR updates all python code to use timezone.utc and ZoneInfo instead of pytz for better compatibility, consistency and speed.

Arrow is next 😉

### CODE REVIEW

File: parsers/ELEXON.py [hunk 1]
```diff
@@ -11,13 +11,12 @@
 """
 
 import re
-from datetime import date, datetime, time, timedelta
+from datetime import date, datetime, time, timedelta, timezone
 from io import StringIO
```
Comment (wobniarin) [inline parsers/ELEXON.py:15]: More a curiosity/learning for me than a change request: Why are we using `timezone` here? Shouldn't it make sense to import ZoneInfo as we do for the other zones and get `TIMEZONE = ZoneInfo("Europe/London")` from there?

**Review by wobniarin — COMMENTED** [review]
very straightforward! I only have a question regarding when to use `ZoneInfo` and when to use `timezone.utc`. I will approve afterwards :)

File: parsers/ELEXON.py (hunk 1 above)
Comment (VIKTORVAV99) [reply parsers/ELEXON.py:15]: timezone from datetime have a singleton with a static definition for utc, this is the recommended approach by the python documentation.

**Comment by VIKTORVAV99** [comment]
> very straightforward! I only have a question regarding when to use `ZoneInfo` and when to use `timezone.utc`. I will approve afterwards :)  

Basically we should use timezone.utc whenever it's UTC (only) and ZoneInfo otherwise. (As per the python documentation)

**Comment by wobniarin** [comment]
> > very straightforward! I only have a question regarding when to use `ZoneInfo` and when to use `timezone.utc`. I will approve afterwards :)
> 
> Basically we should use timezone.utc whenever it's UTC (only) and ZoneInfo otherwise. (As per the python documentation)

Thanks a lot for the explanation! Are we planning on documenting it somewhere so that contributors can learn these best practices? I think we should also share them with the team!

**Review by wobniarin — APPROVED** [review]

**Comment by VIKTORVAV99** [comment]
> > > very straightforward! I only have a question regarding when to use `ZoneInfo` and when to use `timezone.utc`. I will approve afterwards :)
> > 
> > 
> > Basically we should use timezone.utc whenever it's UTC (only) and ZoneInfo otherwise. (As per the python documentation)
> 
> Thanks a lot for the explanation! Are we planning on documenting it somewhere so that contributors can learn these best practices? I think we should also share them with the team!

I'll make sure our wiki pages are updated and I can write together a small email and send that out to engineering tomorrow.

**Pull request merged**